rules/mapping.go

```diff
@@ -69,6 +73,50 @@ func newMappingRuleSnapshot(
 	}, nil
 }
 
+type mappingRuleSnapshotJSON struct {
```

Comment:
those structs are only supposed to be used by a ruleset. They don't need to be exposed.